pkg/pillar/cmd/zedagent/handlecertconfig.go

```diff
@@ -200,15 +207,14 @@ func sendCertsProtobuf(attestReq *attest.ZAttestReq, iteration int) {
 	_, _, rtf, err := zedcloud.SendOnAllIntf(&zedcloudCtx, attestURL,
 		size, buf, iteration, return400)
 	if err != nil {
-		// Hopefully next timeout will be more successful
 		if rtf == types.SenderStatusRemTempFail {
 			log.Errorf("sendCertsProtobuf remoteTemporaryFailure: %s",
 				err)
 		} else {
 			log.Errorf("sendCertsProtobuf failed: %s", err)
 		}
-		return
-	} else {
-		writeSentCertsProtoMessage(data)
+		// sendfailed
+		return false
```

Comment:
This is not the pattern we use. We call zedcloud.SetDeferred, which will make the existing logic retry it until success. Why do you need to invent a different pattern?

Reply:
sure, missed the point. Have made the changes and committed the patch.